janus_server/src/datastore.rs

```diff
@@ -898,6 +898,142 @@ impl Transaction<'_> {
 
         Ok(collect_job_id)
     }
+
+    /// Store a new `batch_aggregations` row in the datastore.
+    #[cfg(test)]
+    #[tracing::instrument(skip(self), err)]
+    pub(crate) async fn put_batch_aggregation<A>(
+        &self,
+        task_id: TaskId,
+        batch_interval_start: Time,
+        aggregate_share: &A,
+        report_count: u64,
+        checksum: &[u8],
```

Comment:
Might be nice to have a `Checksum` newtype around `[u8; 32]` but this PR is already too big.